### PR TITLE
Change all bug ID to links

### PR DESCRIPTION
**Pull request by Geod24**
Context: There has been work done to move towards Github, but concerns have been raised that it would make issue number ambiguous. In order to avoid this undesirable side effect, this PR converts most issue number (I might have missed a few, but I doubt that there's many) into HTTP links.

See also: https://github.com/dlang/projects/issues/43
CC @CyberShadow @WalterBright

### CODE REVIEW

**Comment by CyberShadow** [comment]
> and I do believe the same reasoning (clickable, unmistakable references) applies here as well.

Good point.

**Comment by Geod24** [comment]
Fixed the style issue.
Indeed should have mentioned the precedent set by Walter: https://github.com/dlang/dmd/pull/6775

**Comment by Geod24** [comment]
So... Can we move forward with this before it becomes stale ?

**Comment by CyberShadow** [comment]
Thank you! This looked like a lot of busywork.

File: std/algorithm/iteration.d [hunk 1]
```diff
@@ -2402,7 +2412,9 @@ version (none) // this example requires support for non-equivalence relations
         assert(!range.empty);    // Opposite of refInputRange test
     }
 
-    /* Issue 93532 - General behavior of non-forward input ranges.
+    /* https://issues.dlang.org/show_bug.cgi?id=93532
```
Comment (schveiguy) [inline std/algorithm/iteration.d:2415]: We don't have 90k issues yet, right? ;)

File: std/algorithm/iteration.d [hunk 2]
```diff
@@ -2517,7 +2529,7 @@ version (none) // this example requires support for non-equivalence relations
         }
     }
 
-    // Issue 93532 - Using roundRobin/chunkBy
+    // https://issues.dlang.org/show_bug.cgi?id=93532 - Using roundRobin/chunkBy
```
Comment (schveiguy) [inline std/algorithm/iteration.d:2532]: ditto and there's a lot more.

File: std/algorithm/iteration.d (hunk 1 above)
Comment (MoonlightSentinel) [reply std/algorithm/iteration.d:2415]: Real issue: 19532

File: std/algorithm/iteration.d (hunk 2 above)
Comment (MoonlightSentinel) [reply std/algorithm/iteration.d:2532]: Real issue: 19532

File: std/algorithm/mutation.d [hunk 3]
```diff
@@ -1998,7 +2004,7 @@ private auto removeImpl(SwapStrategy s, Range, Offset...)(Range range, Offset of
     import std.exception : assertThrown;
     import std.range;
 
-    // http://d.puremagic.com/issues/show_bug.cgi?id=10173
+    // https://issues.dlang.org/show_bug.cgi?id=10173
```
Comment (schveiguy) [inline std/algorithm/mutation.d:2007]: Hah, you even caught these! nice.

File: std/algorithm/iteration.d (hunk 1 above)
Comment (Geod24) [reply std/algorithm/iteration.d:2415]: Thanks! I did see it, but couldn't find the real issue, and then forgot about it... Will fix.

**Comment by Geod24** [comment]
Added a commit to correct the issue number, thanks @MoonlightSentinel

File: std/container/rbtree.d [hunk 4]
```diff
@@ -2185,14 +2186,15 @@ if ( is(typeof(binaryFun!less((ElementType!Stuff).init, (ElementType!Stuff).init
     assert(rt1.upperBound(2).equal([3, 4, 5]));
 }
 
-// issue 15941
+// =https://issues.dlang.org/show_bug.cgi?id=issue 15941
```
Comment (schveiguy) [inline std/container/rbtree.d:2189]: invalid link

File: std/algorithm/mutation.d (hunk 3 above)
Comment (Geod24) [reply std/algorithm/mutation.d:2007]: Looooooots of manual review

**Comment by Geod24** [comment]
@schveiguy : Thanks for doing that tedious review! LMK when you're done, I'll fix and reapply `auto-merge`.

File: std/conv.d [hunk 5]
```diff
@@ -5485,7 +5495,8 @@ version (StdUnittest)
     }
 }
 
-@safe unittest //@@@9559@@@
+// https://issues.dlang.org/show_bug.cgi?id=9959
```
Comment (schveiguy) [inline std/conv.d:5498]: typo

File: std/conv.d [hunk 6]
```diff
@@ -5722,10 +5733,6 @@ pure nothrow @safe /* @nogc */ unittest
     assert(n.refs == 10);
 }
 
-@system unittest
-{
-}
```
Comment (schveiguy) [inline std/conv.d:5727]: lol

**Comment by CyberShadow** [comment]
I suggest sending follow-up fixes as new PRs to avoid retriggering the CIs and increasing the chances of merge failures.

**Comment by schveiguy** [comment]
I agree, just merge this. I'm 37/69 files reviewed. Anything I find can be fixed in a smaller update.

> Thanks for doing that tedious review!

Some of it github makes easy by highlighting just the changed portions of a line, and of course being able to mark files as viewed. But this doesn't compare to the tediousness of actually making this fix. Thanks!

File: std/internal/cstring.d [hunk 7]
```diff
@@ -200,7 +201,7 @@ nothrow @nogc @system unittest
     assert(tempCString(abc[].byChar)[] == abc);
 }
 
-// Bugzilla 14980
+// https://issues.dlang.org/show_bug.cgi?id=14980
```
Comment (schveiguy) [inline std/internal/cstring.d:204]: Note check this and the other change against git blame. 1498 and 14980 are a suspicious coincidence.

File: std/traits.d [hunk 8]
```diff
@@ -1498,7 +1501,6 @@ if (func.length == 1 && isCallable!func)
                     // like this.
                     auto " ~ val ~ " = " ~ args ~ "[0];
                     auto " ~ ptr ~ " = &" ~ val ~ ";
-                        // workaround Bugzilla 16582
```
Comment (schveiguy) [inline std/traits.d:1501]: Why was this removed?

Comment (Geod24) [reply std/traits.d:1501]: It's already mentioned 7 lines above (expand the diff upwards)

File: std/algorithm/iteration.d [hunk 9]
```diff
@@ -3941,7 +3956,8 @@ The number of seeds must be correspondingly increased.
 
 @safe unittest
 {
-    // Issue #10408 - Two-function reduce of a const array.
+    // https://issues.dlang.org/show_bug.cgi?id= 10408
```
Comment (schveiguy) [inline std/algorithm/iteration.d:3959]: Missed this the first time -- invalid link.

File: std/traits.d (hunk 8 above)
Comment (schveiguy) [reply std/traits.d:1501]: Ah ok, I did expand but didn't see it. My brain is numbed by this review lol.

**Comment by schveiguy** [comment]
A recommendation for future similar PRs, to make this much easier to review -- One commit that simply replaces the number with the link. No format editing, no removals of outdated anything. Just a simple replacement. This enables one to review simply that the numbers are not changed (that was my main focus). Github makes this easy by highlighting the only the changed portion of the text. Then more commits to fix the formatting, etc.

File: std/internal/cstring.d (hunk 7 above)
Comment (Geod24) [reply std/internal/cstring.d:204]: https://issues.dlang.org/show_bug.cgi?id=14980
```
commit 156bf68ef7008343755ab450facdd0e5604f6fac
Author: k-hara <k.hara.pg@gmail.com>
Date:   Mon Aug 31 21:57:01 2015 +0900

    fix Issue 14980 - getAddressInfo(null) broken
```
156bf68ef7008343755ab450facdd0e5604f6fac

**Comment by Geod24** [comment]
Right. I originally didn't expect it to be so big, but after the initial (scripted) madness, I realized that the style was wildly inconsistent (e.g. references used as well as how things were organized) and had to go in manually and fix tons of stuff.
For reference, there was an unavoidable amount of stylistic changes needed, because the style linter complains about long lines (triggered by `@some_attributes unittest // here goes the long URL and some description`) and some patterns, e.g. it enforces allman but relies on `{` being at EOL to allow `void foo() { }`. But this also allow `void foo() { // Oops`.

But point taken, I'll keep the future changes much shorter.

**Comment by Geod24** [comment]
Followed up on: https://github.com/dlang/phobos/pull/7446
Thanks for the careful review!

**Comment by schveiguy** [comment]
You may misunderstand, I mean separate the adding of the urls from the formatting in 2 *commits*, not 2 *PRs*. This shouldn't affect the style checker.

File: std/internal/cstring.d (hunk 7 above)
Comment (schveiguy) [reply std/internal/cstring.d:204]: This is what I meant: https://github.com/dlang/phobos/pull/7079/files

The prior change was accidentally changed to 1498 instead of 14980 when it was moved. Please fix in the new PR.
